File: js/7kyu/maximum_length_difference.js
// https://www.codewars.com/kata/5663f5305102699bad000056/train/javascript

function mxdiflg(a1, a2) {

  let max_len = 0;

  if (a1.length === 0 || a2.length === 0) {
    return -1;
  }

  a1.forEach(w1 => {
    a2.forEach(w2 => {
      let diff = Math.abs(w1.length - w2.length);
      if (diff > max_len) {
        max_len = diff;
      }
    })
  });

  return max_len;

}



var s1 = ["hoqq", "bbllkw", "oox", "ejjuyyy", "plmiis", "xxxzgpsssa", "xxwwkktt", "znnnnfqknaz", "qqquuhii", "dvvvwz"];
var s2 = ["cccooommaaqqoxii", "gggqaffhhh", "tttoowwwmmww"];

mxdiflg(s1, s2)

File: python/7kyu/maximum_length_difference.py
def mxdiflg(a1, a2):

  if (a1 == [] or a2 == []):
    return -1

  max_len = 0
  for w1 in a1:
    for w2 in a2:
      diff = abs(len(w2) - len(w1))
      if (diff > max_len):
        max_len = diff

  return max_len




s1 = ["hoqq", "bbllkw", "oox", "ejjuyyy", "plmiis", "xxxzgpsssa", "xxwwkktt", "znnnnfqknaz", "qqquuhii", "dvvvwz"]
s2 = ["cccooommaaqqoxii", "gggqaffhhh", "tttoowwwmmww"]


print(mxdiflg(s1,s2))